### PR TITLE
Move assigner runner to separate library and add test for it

### PR DESCRIPTION
**Pull request by makxenov**

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -6,4 +6,5 @@ find_package(GTest REQUIRED)
 
 include(GoogleTest)
 
+add_subdirectory(bin)
```
Comment (akokoshn) [inline tests/CMakeLists.txt:9]: not sure i make sense add `bin` directory, may be locate `test/runner`?

File: tests/bin/assigner/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+add_executable(assigner_test assigner_test.cpp)
```
Comment (akokoshn) [inline tests/bin/assigner/CMakeLists.txt:1]: This test should check `runner` but not `assigner`, so better name `runner_test`

File: tests/bin/assigner/assigner_test.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                                                                30   // selectors
+                                                            }};
+
+    // initialize assignment tables
+    std::vector<nil::blueprint::assignment<ArithmetizationType>> assignments;
+    auto res = run_assigner(input_block, STATE_CONFIG, boost::log::trivial::severity_level::debug, column_sizes, assignments);
+    ASSERT_TRUE(res);
+    ASSERT_EQ(res.value(), EVMC_SUCCESS);
+    ASSERT_EQ(assignments.size(), 2);
+    ASSERT_EQ(assignments[0].witness(0,0), 0);
```
Comment (akokoshn) [inline tests/bin/assigner/assigner_test.cpp:50]: make sense check values which are not 0

File: libs/assigner_runner/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+target_link_libraries(zkEVMAssignerRunner
+                        PUBLIC
+                        evmone_assigner
+                        zkEVMDataTypes
+                        zkEVMJsonHelpers
+                        zkEVMOutputArtifacts
+                        Boost::log
+                        PRIVATE
+                        evmone
+)
```
Comment (aleasims) [inline libs/assigner_runner/CMakeLists.txt:26]: Should we install this library here?

Comment (makxenov) [reply libs/assigner_runner/CMakeLists.txt:26]: Thanks, added

File: libs/assigner_runner/include/zkevm_framework/assigner_runner/runner.hpp [hunk 5]
```diff
@@ -0,0 +1,78 @@
+#ifndef ZKEMV_FRAMEWORK_BIN_ASSIGNER_INCLUDE_RUNNER_HPP_
```
Comment (aleasims) [inline libs/assigner_runner/include/zkevm_framework/assigner_runner/runner.hpp:1]: Wrong include guard

**Review by aleasims — CHANGES_REQUESTED** [review]

**Review by aleasims — APPROVED** [review]

File: tests/libs/assigner_runner/runner_test.cpp [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    evmc::accounts account_storage;
+    auto init_err = init_account_storage(account_storage, STATE_CONFIG);
+    ASSERT_FALSE(init_err.has_value());
+    single_thread_runner<BlueprintFieldType> runner(account_storage, column_sizes,
+                                                    boost::log::trivial::severity_level::debug);
+    auto run_err = runner.fill_assignments(input_block);
+    ASSERT_FALSE(run_err.has_value());
+    auto assignments = runner.get_assignments();
+    ASSERT_EQ(assignments.size(), 2);
```
Comment (akokoshn) [inline tests/libs/assigner_runner/runner_test.cpp:47]: maybe extra check, we'll have to modify it each time when researchers update logic of filling assignment tables. I guess successful execution enough for now

File: bin/assigner/example_data/call_block.json [hunk 7]
```diff
@@ -32,7 +32,7 @@
             "receive_address" : "0x0000000000000000000000000000000000000003",
             "gas_price" : 3,
             "gas" : 200000,
-            "data" : "0x60006000600060006000600161007df1",
+            "data" : "0x60026003600060006000600161007df1",
```
Comment (akokoshn) [inline bin/assigner/example_data/call_block.json:35]: what exactly changed here?

File: bin/emulator/CMakeLists.txt [hunk 8]
```diff
@@ -17,12 +17,11 @@ target_compile_options(${TARGET_NAME} PRIVATE "-fPIC")
 
 target_link_libraries(${TARGET_NAME}
     PRIVATE
-    evmone
     evmone_assigner
+    zkEVMAssignerRunner
     zkEVMDataTypes
     zkEVMJsonHelpers
-    zkEVMOutputArtifacts
     zkEVMBlockGenerator
```
[the remark this answers is not in the document]
Comment (makxenov) [reply bin/emulator/CMakeLists.txt:24]: Done

File: libs/assigner_runner/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+find_package(Boost COMPONENTS REQUIRED log)
+target_link_libraries(zkEVMAssignerRunner
+                        PUBLIC
+                        evmone_assigner
+                        zkEVMDataTypes
+                        zkEVMJsonHelpers
+                        zkEVMOutputArtifacts
+                        Boost::log
+                        PRIVATE
+                        evmone
```
[the remark this answers is not in the document]
Comment (makxenov) [reply libs/assigner_runner/CMakeLists.txt:26]: Done

File: tests/libs/assigner_runner/runner_test.cpp (hunk 6 above)
Comment (makxenov) [reply tests/libs/assigner_runner/runner_test.cpp:49]: Disabled

File: bin/assigner/example_data/call_block.json (hunk 7 above)
Comment (makxenov) [reply bin/assigner/example_data/call_block.json]: There change appeared to be non-functional, reverted, thanks

**Pull request merged**